server/index.js

```diff
@@ -14,6 +14,11 @@ app.use(bodyParser.json());
 app.use(express.static(publicDir));
 app.use(express.static(distDir));
 
+// The Supar Generic Final Error Middleware
+app.use(function(err, req, res, next) {
```

Comment:
Doesn't seem like you are using all these variables.

Reply:
In order for express to recognize that this is an error middleware, your function signature needs to expect 4 variables even if you aren't using them: https://expressjs.com/en/guide/error-handling.html#writing-error-handlers

I'm not using `next()` because we handle the response here with the error message.